engines/batch-ensembler/README.md

```diff
@@ -0,0 +1,133 @@
+# Turing Batch Ensembler
+
+Pyspark application for running batch ensembling jobs in Turing.
+
+## Usage
+```shell
+usage: main.py [-h] --job-spec <job-spec.yaml> [-l {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
+
+Run a PySpark batch ensembling job
+
+optional arguments:
+  --job-spec  <path/to/job-spec/yaml>                   Path to the ensembling job YAML file specification
+  --log-level <DEBUG||INFO||WARNING||ERROR||CRITICAL>   Set the logging level
+  -h, --help                                            show this help message and exit
+```
+
+### Job Specification
+Application expects to receive a path of a job specification via `--job-spec` argument.
+Job specification contains the information on how to configure data source and sink for 
+the job, as well as the user-defined ensembler configuration. The complete schema of 
+the job specification is defined at this [proto](./api/proto/v1/batch_ensembling_job.proto).
+
+#### Job Spec example
+```yaml
+version: v1
+kind: BatchEnsemblingJob
+metadata:
+  name: batch-ensembling
+  annotations:
+    spark/spark.jars: "https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-2.0.1.jar"
+    spark/spark.jars.packages: "com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.19.1"
+    hadoopConfiguration/fs.gs.impl: "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem"
+    hadoopConfiguration/fs.AbstractFileSystem.gs.impl: "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS"    
+spec:
+  source:
+    dataset:
+      type: BQ
+      bqConfig:
+        query: |-
+          WITH customer_filter AS (
+            SELECT customer_id, target_date
+            FROM `project.dataset.customers_list`
+            WHERE target_date = DATE("2021-03-15", "Asia/Jakarta")
+          ),
+
+          serving_features AS (
+            SELECT *
+            FROM `project.dataset.customers_features`
+          )
+
+          SELECT
+            serving_features.*,
+          FROM
+            serving_features
+            INNER JOIN customer_filter USING (customer_id)
+        options:
+          viewsEnabled: "true"
+          materializationDataset: dataset
+    joinOn:
```

Comment:
@romanwozniak Sorry to bring this up late, the proto defines it as join_on here: https://github.com/gojek/turing/blob/main/engines/batch-ensembler/api/proto/v1/batch_ensembling_job.proto#L80

Reply:
https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html handles both CamelCase and snake_case key formats, so it's the same as `join_on`